java-symbolic-execution/src/test/files/se/Spring5Assert.java

```diff
@@ -0,0 +1,160 @@
+package symbolicexecution.behaviorcache;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+class Spring5Assert {
+
+  Supplier<String> func = () -> "message";
+
+  void testNotNull(@Nullable Object o) {
+    Assert.notNull(o, () -> "message");
+    o.toString();
+  }
+
+  void testNotNullNc(@Nullable Object o) {
+    o.toString(); // Noncompliant
+  }
+
+  void testNotNull2(@Nullable Object o) {
+    Assert.notNull(o, func);
+    o.toString();
+  }
+
+  void testNotNull2Nc(@Nullable Object o) {
+    o.toString(); // Noncompliant
+  }
+
+  void testHasLength() {
+    String s = null;
+    Assert.hasLength(s, () -> "message");
+    if (s.length() > 1) {
+      System.out.println();
+    }
+  }
+
+  void testHasLengthNc() {
+    String s = null;
+    if (s.length() > 1) { // Noncompliant
+      System.out.println();
+    }
+  }
+
+  void testHasText() {
+    String s = null;
+    Assert.hasText(s, () -> "message");
+    if (s.length() > 1) {
+      System.out.println();
+    }
+  }
+
+  void testHasTextNc() {
+    String s = null;
+    if (s.length() > 1) { // Noncompliant
+      System.out.println();
+    }
+  }
+
+  void testNotEmpty() {
+    Object[] arr = null;
+    Assert.notEmpty(arr, func);
+    Object first = arr[0];
+  }
+
+  void testNotEmptyNc() {
+    Object[] arr = null;
+    Object first = arr[0]; // Noncompliant
+  }
+
+  void testNotEmptyColl() {
+    Collection<String> coll = null;
+    Assert.notEmpty(coll, func);
+    coll.toString();
+  }
+
+  void testNotEmptyCollNc() {
+    Collection<String> coll = null;
+    coll.toString(); // Noncompliant
+  }
+
+  void testNotEmptyMap() {
+    Map<Integer, Integer> map = null;
+    Assert.notEmpty(map, func);
+    map.toString();
+  }
+
+  void testNotEmptyMapNc() {
+    Map<Integer, Integer> map = null;
+    map.toString(); // Noncompliant
+  }
+
+  void testIsInstanceOf() {
+    Class<?> type = null;
+    Assert.isInstanceOf(type, new Object(), () -> "watch out!");
+    type.toString();
+  }
+
+  void testIsInstanceOfNc() {
+    Class<?> type = null;
+    type.toString(); // Noncompliant
+  }
```

Comment:
To be honest I'm not sure about all these tests on triggering NPEs, they are all the same (a null object being dereferenced). Just keep one. Different types does not change anything in practice for the engine.